### PR TITLE
Download invoice as CSV

### PR DESCRIPTION
**Pull request by acourdavault**
## Description
very ugly poc to download an invoice as a csv

Not ready for prod

### CODE REVIEW

**Review by johnwcallahan — COMMENTED** [review]
This is a good feature.

We use `react-csv` for the downloadable Linodes List CSV. Our abstraction of that component already takes care of some CSV injection vulnerabilities, and the API is easy to work with.

I just pushed up a change that makes use of this library instead and added the other fields; seems to work OK.

**Comment by johnwcallahan** [comment]
I made a few changes to the placement of the link. This is now ready for review.

File: packages/manager/src/features/Billing/InvoiceDetail/InvoiceDetail.tsx [hunk 1]
```diff
@@ -144,12 +147,29 @@ export const InvoiceDetail: React.FC<CombinedProps> = (props) => {
...
+                    headers={csvHeaders}
+                    data={items}
+                  />
+                  <Button
+                    buttonType="secondary"
+                    onClick={() => csvRef.current.link.click()}
+                    style={{ marginRight: 8 }}
+                  >
+                    Download CSV
+                  </Button>
```
Comment (codyfinn) [inline packages/manager/src/features/Billing/InvoiceDetail/InvoiceDetail.tsx:165]: Minor issue but the button to download the CSV data is not as tall as the button to download the PDF. Also the CVS button has rounded corners whereas the PDF button has square ones.

**Review by codyfinn — APPROVED** [review]
Left a non-blocking comment

**Comment by johnwcallahan** [comment]
Thanks Cody, I just opened up https://github.com/linode/manager/pull/8057 to address.

**Review by bnussman — APPROVED** [review]
Looks great! 🚀

**Pull request merged**